iotdb/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java

```diff
@@ -370,18 +373,36 @@ private void analyzeMetadataCreate(AstNode astNode) throws MetadataArgsErrorExce
     AstNode paramNode = astNode.getChild(1);
     String dataType = paramNode.getChild(0).getChild(0).getText();
     String encodingType = paramNode.getChild(1).getChild(0).getText();
+    String compressor;
+    int offset = 2;
+    if (paramNode.getChildren().size() > offset) {
+      compressor = paramNode.getChild(2).getChild(0).getText();
+      offset++;
+    } else {
+      compressor = TSFileConfig.compressor;
+    }
     checkMetadataArgs(dataType, encodingType);
-    String[] paramStrings = new String[paramNode.getChildCount() - 2];
+    String[] paramStrings = new String[paramNode.getChildCount() - offset];
+    Map<String, String> props = new HashMap<>(paramStrings.length + 1, 1);
+    String[] kv;
+    for (String param : paramStrings) {
```

Comment:
When is the content of `paramStrings` modified. I could only see the initialization.

Reply:
Have fixed...